GitCommands/Patch/Patch.cs

```diff
@@ -1,54 +1,63 @@
-using System.Text;
+﻿using System;
+using JetBrains.Annotations;
 
 namespace PatchApply
```

Comment:
Please align the location with the namespaces or the location with the namespaces.
Same goes for the tests as well

Reply:
Done.

FYI there are currently 62 types in namespaces that don't reflect folder location. Perhaps this should also be a separate issue we fix across the board in one go.

<details>
<summary>Namespaces not matching file locations</summary>

```text
Solution GitExtensions.sln

    Project GitCommands
      GitCommands\BoolExtensions.cs:3 Namespace does not correspond to file location, should be: 'GitCommands'
      GitCommands\DeconstructionExtensions.cs:4 Namespace does not correspond to file location, should be: 'GitCommands'
      GitCommands\Git\GitBlame.cs:7 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitCommand.cs:5 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitCommandCache.cs:4 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitCommandHelpers.cs:16 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitConvert.cs:3 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitDeleteBranchCmd.cs:6 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitDeleteRemoteBranchesCmd.cs:6 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitGpgController.cs:8 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitItemStatus.cs:4 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitModule.cs:22 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitPushAction.cs:5 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitRef.cs:7 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitRevision.cs:10 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitSubmoduleInfo.cs:5 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitSubmoduleStatus.cs:1 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitSvnCommandHelpers.cs:4 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Git\GitVersion.cs:4 Namespace does not correspond to file location, should be: 'GitCommands.Git'
      GitCommands\Settings\AppSettings.cs:16 Namespace does not correspond to file location, should be: 'GitCommands.Settings'
      GitCommands\Settings\BranchOrdering.cs:1 Namespace does not correspond to file location, should be: 'GitCommands.Settings'
      GitCommands\Settings\SettingsCache.cs:4 Namespace does not correspond to file location, should be: 'GitCommands.Settings'
      GitCommands\StringExtensions.cs:6 Namespace does not correspond to file location, should be: 'GitCommands'
      GitCommands\Utils\EncodingHelper.cs:6 Namespace does not correspond to file location, should be: 'GitCommands.Utils'
      GitCommands\Utils\RFC2047Decoder.cs:4 Namespace does not correspond to file location, should be: 'GitCommands.Utils'
    
    Project GitExtUtils
      GitExtUtils\GitUI\UIExtensions.cs:5 Namespace does not correspond to file location, should be: 'GitExtUtils.GitUI'
      GitExtUtils\Linq\LinqExtensions.cs:3 Namespace does not correspond to file location, should be: 'GitExtUtils.Linq'
    
    Project GitUI
      GitUI\CommandsDialogs\BrowseDialog\FormBrowseMenus.cs:9 Namespace does not correspond to file location, should be: 'GitUI.CommandsDialogs.BrowseDialog'
      GitUI\CommandsDialogs\SettingsDialog\EditorHelper.cs:4 Namespace does not correspond to file location, should be: 'GitUI.CommandsDialogs.SettingsDialog'
      GitUI\CommandsDialogs\SettingsDialog\Plugins\PluginRootIntroductionPage.cs:4 Namespace does not correspond to file location, should be: 'GitUI.CommandsDialogs.SettingsDialog.Plugins'
      GitUI\Editor\ColorHelper.cs:4 Namespace does not correspond to file location, should be: 'GitUI.Editor'
      GitUI\Editor\FindAndReplaceForm.cs:12 Namespace does not correspond to file location, should be: 'GitUI.Editor'
      GitUI\Editor\RichTextBoxXhtmlSupportExtension.cs:14 Namespace does not correspond to file location, should be: 'GitUI.Editor'
      GitUI\Plugin\GitPluginSettingsContainer.cs:5 Namespace does not correspond to file location, should be: 'GitUI.Plugin'
      GitUI\Plugin\LoadPlugins.cs:8 Namespace does not correspond to file location, should be: 'GitUI.Plugin'
      GitUI\Plugin\PluginExtraction.cs:7 Namespace does not correspond to file location, should be: 'GitUI.Plugin'
      GitUI\UserControls\BlameControl.cs:11 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\BranchComboBox.cs:10 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\FileStatusList.cs:19 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\GravatarControl.cs:11 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\HelpImageDisplayUserControl.cs:7 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\MenuStripEx.cs:4 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\PatchGrid.cs:8 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\PathFormatter.cs:9 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\RevisionGrid.cs:32 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\RevisionGridClasses\DvcsGraph.cs:13 Namespace does not correspond to file location, should be: 'GitUI.UserControls.RevisionGridClasses'
      GitUI\UserControls\RevisionGridClasses\DvcsGraph.Graph.cs:7 Namespace does not correspond to file location, should be: 'GitUI.UserControls.RevisionGridClasses'
      GitUI\UserControls\RevisionGridClasses\DvcsGraph.Junction.cs:4 Namespace does not correspond to file location, should be: 'GitUI.UserControls.RevisionGridClasses'
      GitUI\UserControls\RevisionGridClasses\DvcsGraph.Lanes.cs:7 Namespace does not correspond to file location, should be: 'GitUI.UserControls.RevisionGridClasses'
      GitUI\UserControls\RevisionGridClasses\FormRevisionFilter.cs:4 Namespace does not correspond to file location, should be: 'GitUI.UserControls.RevisionGridClasses'
      GitUI\UserControls\ToolStripEx.cs:4 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
      GitUI\UserControls\WarningToolStripItem.cs:5 Namespace does not correspond to file location, should be: 'GitUI.UserControls'
    
    Project GitUIPluginInterfaces
      Plugins\GitUIPluginInterfaces\Settings\BoolSetting.cs:4 Namespace does not correspond to file location, should be: 'GitUIPluginInterfaces.Settings'
      Plugins\GitUIPluginInterfaces\Settings\ChoiceSetting.cs:5 Namespace does not correspond to file location, should be: 'GitUIPluginInterfaces.Settings'
      Plugins\GitUIPluginInterfaces\Settings\NumberSetting.cs:3 Namespace does not correspond to file location, should be: 'GitUIPluginInterfaces.Settings'
      Plugins\GitUIPluginInterfaces\Settings\PasswordSetting.cs:3 Namespace does not correspond to file location, should be: 'GitUIPluginInterfaces.Settings'
      Plugins\GitUIPluginInterfaces\Settings\StringSetting.cs:4 Namespace does not correspond to file location, should be: 'GitUIPluginInterfaces.Settings'
    
    Project SpellChecker
      NetSpell.SpellChecker\Properties\AssemblyInfo.cs:38 Namespace does not correspond to file location, should be: 'NetSpell.SpellChecker.Properties'
      NetSpell.SpellChecker\TraceWriter.cs:7 Namespace does not correspond to file location, should be: 'NetSpell.SpellChecker'
    
    Project TfsIntegration
      Plugins\BuildServerIntegration\TfsIntegration\ITfsHelper.cs:5 Namespace does not correspond to file location, should be: 'TfsIntegration'
      Plugins\BuildServerIntegration\TfsIntegration\Properties\AssemblyInfo.cs:38 Namespace does not correspond to file location, should be: 'TfsIntegration.Properties'
    
    Project TfsInterop.Vs2015
      Plugins\BuildServerIntegration\TfsInterop.Vs2015\TfsHelper2015.cs:14 Namespace does not correspond to file location, should be: 'TfsInterop.Vs2015'
```

</details>

Reply:
Yes, I'm acutely aware of these. But given that you're currently making significant changes in this particular area I thought it would be a good time to align this folder/namespace.

Reply:
Equally happy for the namespace alignment to happen as a separate PR